src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,47 @@
+@import "./../utils/variables";
+
+.calendar {
+
+  @each $day, $margin in $days {
+    &-start-day--#{$day} &__days:first-child {
+      margin-left: $margin;
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &-month-length--#{$index} &__days:nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+
+  &__days {
+    display: flex;
+    flex-wrap: wrap;
+
+    width: $sizeBlock;
+    height: $sizeBlock;
+
+    background-color: #eee;
+    border: 1px solid black;
+
+    transition: 0.5s;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        width: 100%;
+        height: 100%;
+        text-align: center;
```

Comment:
add all these styles outside the loop, leave in the loop styles that depend only on the iteration of the loop